### PR TITLE
refactor: migrate tracing core from boot-start to dubbo deployer

### PR DESCRIPTION
**Pull request by conghuhu**
## What is the purpose of the change

The previous Dubbo Tracing implementation overly relied on the spring boot starter. While providing users with better extensibility, Dubbo Tracing also loses its independence.

So I migrated tracing core from boot-start to dubbo deployer.

Child Task of #12448 

## Brief changelog

- Add `dubbo-tracing` module.
- Migrated OpenTelemetry tracer, Brave will migrate in the next PR.
- Transfer zipkin and otlp exporters in OpenTelemetry. Brave's zipkin will be migrated in the next PR.

#### Use SpringBoot Starter
```xml
        <dependency>
            <groupId>org.apache.dubbo</groupId>
            <artifactId>dubbo-spring-boot-tracing-otel-zipkin-starter</artifactId>
        </dependency>
```

#### Use Dubbo directly
```xml
        <!-- dubbo tracing & tracing bridge & exporter -->
        <dependency>
            <groupId>org.apache.dubbo</groupId>
            <artifactId>dubbo-tracing</artifactId>
            <version>${project.version}</version>
        </dependency>
        <dependency>
            <groupId>io.micrometer</groupId>
            <artifactId>micrometer-tracing-bridge-otel</artifactId>
        </dependency>
        <dependency>
            <groupId>io.opentelemetry</groupId>
            <artifactId>opentelemetry-exporter-zipkin</artifactId>
        </dependency>
```

## Verifying this change

Local test result:

d558ce91d92e234a7fba5b7e81a7f6b
672b23c8255b5a48fcdef30d2bd235d

dubbo-samples：[https://github.com/apache/dubbo-samples/pull/854](https://github.com/apache/dubbo-samples/pull/854)

## TODO

- Add Non SpringBoot starter Sample in `dubbo-samples`.
- Migrate Brave.

## Checklist
- [x] Make sure there is a [GitHub_issue](https://github.com/apache/dubbo/issues) field for the change (usually before you start working on it). Trivial changes like typos do not require a GitHub issue. Your pull request should address just this issue, without pulling in other changes - one PR resolves one issue.
- [ ] Each commit in the pull request should have a meaningful subject line and body.
- [ ] Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
- [ ] Check if is necessary to patch to Dubbo 3 if you are work on Dubbo 2.7
- [ ] Write necessary unit-test to verify your logic correction, more mock a little better when cross module dependency exist. If the new feature or significant change is committed, please remember to add sample in [dubbo samples](https://github.com/apache/dubbo-samples) project.
- [ ] Add some description to [dubbo-website](https://github.com/apache/dubbo-website) project if you are requesting to add a feature.
- [ ] GitHub Actions works fine on your own branch.
- [ ] If this contribution is large, please follow the [Software Donation Guide](https://github.com/apache/dubbo/wiki/Software-donation-guide).

### CODE REVIEW

**Comment by conghuhu** [comment]
@songxiaosheng @AlbumenJ @chickenlj PTAL

**Review by AlbumenJ — CHANGES_REQUESTED** [review]
Mode ObservationSenderFilter  to dubbo-tracing

File: dubbo-common/src/main/java/org/apache/dubbo/config/nested/ExporterConfig.java [hunk 1]
```diff
@@ -56,14 +55,27 @@ public static class ZipkinConfig implements Serializable {
         private String endpoint;
 
         /**
-         * Connection timeout for requests to Zipkin.
+         * Connection timeout for requests to Zipkin. (seconds)
          */
-        private Duration connectTimeout = Duration.ofSeconds(1);
+        private Integer connectTimeout = 1;
```
Comment (AlbumenJ) [inline dubbo-common/src/main/java/org/apache/dubbo/config/nested/ExporterConfig.java:60]: Is it possible not to change this signature?

**Review by AlbumenJ — CHANGES_REQUESTED** [review]

**Comment by conghuhu** [comment]
> ObservationSenderFilter
image

I've finshed to move ObservationSenderFilter @AlbumenJ

**Review by songxiaosheng — APPROVED** [review]
LGTM

**Review by songxiaosheng — APPROVED** [review]
LGTM

File: dubbo-config/dubbo-config-api/src/main/java/org/apache/dubbo/config/deploy/DefaultApplicationDeployer.java [hunk 2]
```diff
@@ -409,26 +415,35 @@ private void initMetricsReporter() {
...
+            return;
+        }
+        Optional<TracingConfig> configOptional = configManager.getTracing();
+        if (!configOptional.isPresent() || !configOptional.get().getEnabled()) {
+            return;
+        }
 
-    private static boolean isClassPresent(String className) {
-        return ClassUtils.isPresent(className, DefaultApplicationDeployer.class.getClassLoader());
+        DubboObservationRegistry dubboObservationRegistry = new DubboObservationRegistry(applicationModel, configOptional.get());
```
Comment (chickenlj) [inline dubbo-config/dubbo-config-api/src/main/java/org/apache/dubbo/config/deploy/DefaultApplicationDeployer.java:439]: So only when all the above three conditions are matched will we initiate DubboObservationRegistry?

Comment (conghuhu) [reply dubbo-config/dubbo-config-api/src/main/java/org/apache/dubbo/config/deploy/DefaultApplicationDeployer.java:439]: Yes, due to the Observation.class was proposed after version 1.10 of micrometer, when users use springroot2.x, it defaults to 1.9.x, which means Observation.class is missing. We have made the corresponding judgment and provided users with debug level log prompts.

As for Option<TracingConfig>, this is the switch that determines whether Dubbo itself has enabled the Tracing function

**Review by AlbumenJ — APPROVED** [review]

**Comment by AlbumenJ** [comment]
`dubbo_provider_requests_total` is missing after this PR merged. Revert in [ea35f7ed5c1d6d2a6ede821f1481eb5088254d28](https://github.com/apache/dubbo/commit/ea35f7ed5c1d6d2a6ede821f1481eb5088254d28)

**Pull request merged**